File: backend/src/main/java/finki/ukim/backend/administration/service/domain/MunicipalityService.java
package finki.ukim.backend.administration.service.domain;

import finki.ukim.backend.administration.model.domain.Municipality;
import finki.ukim.backend.administration.model.dto.filters.MunicipalityFilterDto;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Optional;

public interface MunicipalityService {
    List<Municipality> findAll();

    Municipality findById(Long id);

    Municipality findByCode(String code);

    Municipality create(Municipality municipality);

    Municipality update(Long id, Municipality municipality);

    Municipality deleteById(Long id);

    void deleteAllById(List<Long> ids);

    Page<Municipality> findAll(MunicipalityFilterDto municipalityFilterDto);
}

File: frontend/src/administration/service/municipalityService.js
import axiosInstance from "../../common/axios/axios.js";

const municipalityService = {
    findAll: async () => {
        return await axiosInstance.get("/municipalities");
    },

    findAllPaged: async ({
                             page = 0,
                             size = 10,
                             sortBy = "id",
                             id = null,
                             code = null,
                             name = null
                         }) => {
        return await axiosInstance.get("/municipalities/paged", {
            params: {
                page,
                size,
                sortBy,
                id,
                code,
                name
            }
        });
    },

    findById: async (id) => {
        return await axiosInstance.get(`/municipalities/${id}`);
    },

    findByCode: async (code) => {
        return await axiosInstance.get(`/municipalities/by-code/${code}`);
    },

    create: async (data) => {
        return await axiosInstance.post("/municipalities", data);
    },

    update: async (id, data) => {
        return await axiosInstance.put(`/municipalities/${id}`, data);
    },

    deleteById: async (id) => {
        return await axiosInstance.delete(`/municipalities/${id}`);
    }
};

export default municipalityService;